compiler/luci/profile/include/luci/Profile/ProfilingData.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __LUCI_PROFILING_DATA_H__
+#define __LUCI_PROFILING_DATA_H__
+
+#include <loco.h>
+
+namespace luci
+{
+
+/**
+ * @brief Set annotation for circle node id
+ * @note  Once NodeID is annotated, it should not be changed.
+ *        If NodeID is needed to be changed, create new CircleNodeID.
+ */
+class CircleNodeID final : public loco::NodeAnnotation
+{
+public:
+  CircleNodeID() = delete;
+
+  CircleNodeID(uint32_t node_id) : _node_id{node_id}
+  {
+    // Do nothing
+  }
+
+public:
+  uint32_t node_id(void) const { return _node_id; }
+  // void node_id(uint32_t id) { _node_id = id; }
```

Comment:
is this needed?

Reply:
Just for letting know users that setter is not implemented intentionally.

Then `// No setter` or just removing is better?

Reply:
> Then // No setter or just removing is better?

Yes, that is explict expression and I think it's better.